### ISSUE TITLE
Don't auto close `?private-chat` created by `moderators`

### ISSUE THREAD

**Issue by smeijer** [kentcdodds/kcd-discord-bot-v1#25]
**Problem description**:

Moderators can open a private chat with a user, to talk about issues. Because private chats are being closed after a few minutes of inactivity, it's possible that the chat is being closed before the invited user reads the message. Resulting in the message never being read.

**Suggested solution**:

When a moderator starts a private chat, the chat should not be closed automatically. Instead, the moderator should request the bot explicitly to close & remove it.

**Comment by kentcdodds** [comment]
This is a great point. I don't think the rule should be "if it's created by a mod, don't auto-close it" though, because mods may create private chats for other reasons as well. I think we just need better control over how auto-closures work. Especially in regards to triaging CoC issues. Like, maybe we'll give the user 24 hours to acknowledge they've seen the message and if they don't then the bot will ping them again about it. If they don't acknowledge it again, then the bot can kick them (they can rejoin if they decide to later).

Just an idea. There are a few things to consider here... I'll noodle on it.

**Comment by smeijer** [comment]
> Like, maybe we'll give the user 24 hours to acknowledge...

That might not be a bad idea at all. It makes me even consider a more "extreme" way of moderation, that's a bit less personal.

Imagine, the following scenario:

```
user: posts CoC violating message
mod:  flags message
bot:  removes post & kicks user

# user now only sees a single channel, in which they need to reply to the bot they understand the CoC.

user: acknowledges
bot:  restores user access
```

That sounds extreme. But... this way `user` won't have hard feelings to a `mod`. They also doesn't need to spend "10 minutes" talking to a `mod` explaining the situation. And they can't do more harm, while the `mod` is awaiting response. All they need to do is tell the `bot` they want back in. 

This can later be expanded with `cool-downs`. Think of:

```
kick 1: access restores directly
kick 2: access restores after 15 minutes
kick 3: ... 60 minutes
kick 4: ... 24 hours
kick 5: ... 72 hours
kick 6: ban
```

And in addition to the 1 bot channel, a second `talk to mods` channel could be added. So that the user has a way to talk to the mods in case they think they're treated unfairly.  

Needless to say, this should only be used for clear violations! Not for an innocent misplaced meme that shouldn't have been posted.
 
Just another thought I found worth sharing.

**Comment by kentcdodds** [comment]
I'm a fan of all the ideas you're sharing @smeijer 👍 And yes, this would only be for actual CoC violations.

For some small stuff like misplaced memes/posts, what if a moderator could add a reaction to the post which would trigger the bot to delete it and send the user a private message just to let them know that they're not in trouble or anything, the post was just deleted because some may find it offensive. Thoughts?

**Comment by smeijer** [comment]
Sounds good to me. So a `delete + private message` for small stuff, and a `delete + kick + re-acknowledge understanding of CoC` for clear violations of the CoC.

I think we can fix those with commands that can be posted in the `#🔒-moderators-room `, and I would also propose to make the commands only effective there. That way other mods know what's going on, and users don't.

How about:

```
?delete {link-to-message}
?kick {link-to-message}
```

* edit, I first thought about `?warn` and `?kick`, but as it's not really an official warning, we might want to keep that one for the future. `?delete` covers it, and the user is just being informed about the action.

Alternatively, it might be possible to flag things with reactions:

`🗑️` for `?delete`
`🚩` for `?kick`

reactions have the benefit that it's easier to manage on the phone.

**Comment by kentcdodds** [comment]
Yeah, I think let's do reactions rather than commands and the bot will just post to the moderators channel when an action like that is taken.

To be clear, `kick` in discord terms means the user is no longer a member of the server and has to rejoin. I was thinking "penalty box" is more like what that is, but I don't like how punitive that sounds 😅 It's also kinda like a "time out" but that also sounds punitive 🙃 Not sure how to refer to this, but yeah, I think emoji would probably be best.

**Comment by smeijer** [comment]
In discord terms, `kick` might not be correct. But as I'm being kicked out of the conversation, I think I would still experience it that way.

I'm getting old here. I'm not familiar with discord naming, so I compare it a bit with IRC. There are also servers and channels, and kicking happens on a channel level there. 

At the end, it's the common developer problem. Naming things. I'm okay with whatever you come op with 😅

> Do not pass Go. Do not collect $200

**Comment by MichaelDeBoey** [comment]
I really like the idea of automating all of this so there won't be any negative feelings towards the mods.

I also like the cool-down idea like @smeijer explained.
Although I think we should also have a way of decreasing the cool-down counter.
So if people are reacting in the `talk to mods` channel and the mod decides it was indeed unnecessary to do the kick, the kick will not count for extra cool-down time

I also like the emoji approach.

And of course we need to make sure these actions will only be triggered by mods/admins.
So if non-mods/admins react with the chosen emoji's it won't trigger the kick
